doc/specs/#1595 - Suggestions UI/Snippets.md

```diff
@@ -0,0 +1,526 @@
+  - This will need to be reconciled with [#7039], which tracks displaying
+    non-localized names in the command palette
+  - It won't be a TeachingTip, since those are an unmitigated disaster. But we
+    can just fake it with another text box.
+* [X] [#1595] Add the Suggestions UI, with support for `tasks`
+* [x] Fragments can add **actions** to a user's settings
+
+### 🚶 Walk
+* [ ] The terminal can look for a settings file of tasks in a profile's
+  `startingDirectory` (regardless of shell integration being enabled)
+* [ ] [#5790] - profile specific actions
+* [ ] [#12857] Ability to save selected text as a `sendInput` action
+* [ ] [#12861] Re-evaluate showing some sort of "ghost text or other preview for snippets
+
+### 🏃‍♂️ Run
+* [ ] When the user `cd`s to a directory (with shell integration enabled), the
+  terminal can load the tasks from that directory tree
+* [ ] [#10436] Users can manage all their fragments extensions directly in the Settings UI
+* [ ] The suggestions UI & snippets pane can filter not only on name of a
+  command, but for snippets, the input as well.
+* [ ] [#12927] Enlighten the suggestions UI to support (_a yet undeclared syntax
+  for_) snippets with prompt-able sections in them
+
+<!--
+
+These are perhaps too vague to be included in the official spec, but they are left here for posterity:
+
+### 🚀 Sprint
+
+* [ ] Fork of [#12927] - prompt-able sections can accept a command to dynamically populate options
+* [ ] Enlighten the suggestions UI to understand a yet undeclared syntax for snippets with prompt-able sections in them.
+* [ ] Stand up a dedicated repo for `terminal-snippets`
+  * [ ] Pull those down, or compile them in, or so -->
+
+## Conclusion
+
+Snippets are something that developers immediately understand the value of.
+After talking with users, everyone we talked with immediately understood the
+concept, and you could see the gears turning on ways to integrate this into
+their own workflows.
+
+### Future Considerations
+
+* We may want to add additional params to the `save` subcommand in the future,
+  to configure where the snippet is saved:
+  * `--local`: Save to the `.wt.json` in the CWD, if there is one (or create
+    one)
+  * `--parent`: Save to the `.wt.json` in the first ancestor of the CWD, if
+    there is one. Otherwise create one here.
+  * `--settings`: Manually save to the settings file?
+  * `--profile`: save to this profile???? Not sure if this is actually possible.
+    Maybe with the `WT_SESSION_ID` env var to figure out which profile is in use
+    for the pane with that ID
+* Longer workflows might be better exposed as notebooks. We've already got a
+  mind to support [markdown in a notebook-like
+  experience](https://github.com/microsoft/terminal/issues/16495) in the
+  Terminal. For longer scripts that may need rich markup between commands, that
+  will likely be a better UX.
+* For what it is worth, [Warp] uses .yaml files for their "workflows". As an

```

Comment:
Two slightly different thoughts on importing the Warp YAML:
1. `wt import` is tracked [here](https://github.com/microsoft/terminal/issues/10083). Yeah, it's a different request, but figured it'd be nice to be aware of this.
2. We've discussed [importing a color scheme via the settings UI](https://github.com/microsoft/terminal/issues/8639) in the past. Your idea here is basically the same thing as that, but for snippets. Presumably, snippets will have their own page in the settings UI? Do they need one.

Reply:
(i've linked this to the other two threads)